examples/sunmatrix/band/test_fsunmatrix_band_mod.f90

```diff
@@ -118,39 +119,35 @@ integer(C_INT) function unit_tests() result(fails)
 
     ! Fill identity matrix
     Idata => FSUNBandMatrix_Data(I)
-    do jj = 0, N-1
-      colj => FSUNBandMatrix_Column(I, jj)
-      colj(1) = ONE
+    do jj = 1, N
+      Idata(jj) = ONE
     end do
 
     ! Fill A matrix
     smu   = FSUNBandMatrix_StoredUpperBandwidth(A)
     Adata => FSUNBandMatrix_Data(A)
-    do jj = 0, N-1
-      offset = smu-mu+1 + jj*(smu+ml+1)
-      colj(-mu:ml) => Adata(offset:mu+ml)
-      jstart = merge(-jj, -mu, jj < mu)
-      jend = merge(N-1-jj, ml, jj > N-1-ml)
-      do ii = jstart, jend
-        colj(ii) = jj - ii
+    do jj = 1, N
+      offset = (jj-1)*(smu+ml+1) + smu + 1 ! offset to diagonal
+      istart = merge(-mu, -(jj-1), jj > mu) ! above diagonal
+      iend = merge(N-jj , ml, jj > N - ml)  ! below diagonal
+      do ii = istart, iend
+        Adata(offset+ii) = (jj-1) - ii
       end do
     end do
 
     xdata(0:N-1) => FN_VGetArrayPointer(x)
     ydata(0:N-1) => FN_VGetArrayPointer(y)
```

Comment:
Looks like these lines are causing the CI to fail
```suggestion
    xdata => FN_VGetArrayPointer(x)
    ydata => FN_VGetArrayPointer(y)
```

Reply:
Interestingly, they are causing the compiler to crash! 

```
     xdata(0:N-1) => FN_VGetArrayPointer(x)
 1
internal compiler error: Segmentation fault
```

Reply:
That's probably it, the CI machine is using gcc@5.5 and passes without the bound checking flag but with bounds checking enabled if segfaults.